File: contactsinroomlibraryjava/src/main/java/com/example/contactsinroomlibraryjava/data/ContactDAO.java
package com.example.contactsinroomlibraryjava.data;

import android.database.Cursor;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.contactsinroomlibraryjava.Contact;

import java.util.List;

@Dao
public interface ContactDAO {

    @Insert
    long addContact(Contact contact);

    @Update
    void updateContact(Contact contact);

    @Delete
    void deleteContact(Contact contact);

    @Query("select * from contacts")
    List<Contact> getAllContacts();

    @Query("SELECT * FROM contacts")
    Cursor findAll();

    @Query("select * from contacts where contact_id==:contactId ")
    Contact getContact(long contactId);
}

File: contactsinroomlibrary/src/main/java/com/example/contactsinroomlibrary/data/ContactDAO.kt
package com.example.contactsinroomlibrary.data

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.example.contactsinroomlibrary.Contact

@Dao
interface ContactDAO {
    @Insert
    fun addContact(contact: Contact): Long

    @Update
    fun updateContact(contact: Contact)

    @Delete
    fun deleteContact(contact: Contact)

    @Query("select * from contacts")
    fun getAllContacts(): List<Contact>

    @Query("select * from contacts where contact_id = :contactId")
    fun getContact(contactId: Long): Contact
}